### PR TITLE
Foldable / Traversable

### PR DESCRIPTION
**Pull request by DrBoolean**
Not sure if everything is how we want it yet, but thought i'd create this pull request to make it easier to see the changes.

### CODE REVIEW

**Comment by davidchambers** [comment]
What's the status of this pull request?

**Comment by buzzdecafe** [comment]
it ain't blocking me. it would be nice to have for ramda-fantasy

**Comment by 5outh** [comment]
I'm happy with this, :+1: for merge from me

**Comment by joneshf** [comment]
You had my vote last year ;).

**Comment by DrBoolean** [comment]
Right then let's do this!

**Comment by davidchambers** [comment]
:tada:

**Comment by CrossEye** [comment]
:cake:

File: README.md [hunk 1]
```diff
@@ -169,6 +171,51 @@ or its `constructor` object. The `of` method takes one argument:
...
+
+1. `f` must be a binary function
+
+    1. if `f` is not a function, the behaviour of `reduce` is unspecified.
+    2. The first argument to `f` must be the same type as `x`.
+    3. `f` must return a value of the same type as `x`
+
+1. `x` is the initial accumulator value for the reduction
+
+### Traversable
```
Comment (wayneseymour) [inline README.md]: @drboolean May I have an example of Traversable and Sequence?  Pretty pls?

Comment (DrBoolean) [inline README.md]: Here's a gist of the laws: https://gist.github.com/DrBoolean/c4949a0f4b8f9ba8261f

A quick example would be:
```js
[Promise.of(1), Promise.of(2)].sequence(Promise.of)
//=> Promise([1,2])
```
It turns `[Promise]` to `Promise []`

Comment (wayneseymour) [inline README.md]: Oh cool.  Thanks sir.
Going over the gist now.

**Pull request merged**